lib/fastlane/plugin/wpmreleasetoolkit/helper/glotpress_helper.rb

```diff
@@ -26,12 +26,35 @@ def self.get_translation_status_data(glotpress_url:)
       # @return [Integer] The percentage of the translated strings.
       #
       def self.get_translation_status(data:, language_code:)
-        current = extract_value_from_translation_info_data(data: data, language_code: language_code, status: 'current')
-        fuzzy = extract_value_from_translation_info_data(data: data, language_code: language_code, status: 'fuzzy')
-        untranslated = extract_value_from_translation_info_data(data: data, language_code: language_code, status: 'untranslated')
-        waiting = extract_value_from_translation_info_data(data: data, language_code: language_code, status: 'waiting')
+        # The status is parsed from Glotpress project page.
+        # The row can be identified by the language code and the progress is in the column identified by the class "stats percent".
+        # When the progress is above 90%, a special badge is added.
+        # Because of the way the HTML is organized, this regex matches content spawned on three or four lines
+        # Regex:
+        # ^           : start of a line
+        # \s*         : any space
+        # <strong><a href=".*\/#{language_code}\/default\/"> : This link contains the language code of that line in the HTML table, so it's a reliable match
+        # .*          : any character. The language name should be here, but it can be less reliable than the language code as a match
+        # <\/strong>  : tag closure
+        # \n          : new line
+        # (?:         : match the following. This starts the "morethan90" special badge, which we expect to exist zero or one times (see the closure of this part of the regex).
+        #       \s*         : any space
+        #       <span class="bubble morethan90"> : Start of the special badge
+        #       \d\d\d?%    : 2 or 3 digits and the percentage char
+        #       <\/span>\n  : Special badge closure and new line
+        # )?          : end of the "morethan90" special badge section. Expect this zero or one times.
+        # \s*<\/td>\n : column closure tag. Any space before of it are ok. Expect new line after it.
+        # \s*         : any space
+        # <td class="stats percent"> : This is the tag which can be used to extract the progress
+        # ([0-9]+)    : progress is the first group
+        # %<\/td>     : tag closure
+        regex = "^\\s*<strong><a href=\".*\\/#{language_code}\\/default\\/\">.*<\\/strong>\\n"
+        regex += '(?:\s*<span class="bubble morethan90">\d\d\d?%<\/span>\n)?\s*<\/td>\n\s*<td class="stats percent">([0-9]+)%<\/td>$'
```

Comment:
Seeing that `morethan90` CSS class here, I'm assuming it might only be set to that `span` if the percentage is indeed >90 for that `tr` row? Which could mean that this regex could fail to catch cases where the stats percent <90 for which that CSS class is not included in the HTML?

Reply:
Yeah! Initially I thought the same and implemented a slighter simple regex thinking that I could always get the value in that span, but it's not the case. When the percentage is lower than 90, all the span is missing. You can check the relevant GlotPress code here:
https://github.com/GlotPress/GlotPress-WP/blob/0c395a7a8f37ab3b5ebafd2b239c74392e5177f9/gp-templates/project.php#L100